sstables/compaction.cc

```diff
@@ -462,7 +462,7 @@ class compaction {
         };
     }
 
-    virtual flat_mutation_reader::filter filter_func() const {
+    virtual flat_mutation_reader::filter consider_partition() const {
```

Comment:
To me, the new name seems confusing because it's a verb which describes an action which the returned filter does, not what this method does. `make_partion_filter()` / `partition_filter()` seems better.

It's like having `stop()` which doesn't stop but returns a functor which stops.

Reply:
makes sense.  I'll go with `make_partition_filter`.